src/main/java/com/softserve/academy/spaced/repetition/controller/CardController.java

```diff
@@ -97,31 +99,34 @@
         return new ResponseEntity<>(cardPublicDTO, HttpStatus.OK);
     }
 
-
     @Auditable(action = AuditingAction.CREATE_CARD_VIA_COURSE_AND_DECK)
     @PostMapping(value = "/api/category/{categoryId}/courses/{courseId}/decks/{deckId}/cards")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToDeck(#categoryId, #courseId, #deckId)")
     public ResponseEntity<CardPublicDTO> addCardByCourseAndDeck(@Validated(Request.class) @RequestBody Card card,
                                                                 @PathVariable Long categoryId,
                                                                 @PathVariable Long courseId,
-                                                                @PathVariable Long deckId) {
-        cardService.addCard(card, deckId);
-        Link selfLink = linkTo(methodOn(CardController.class)
-                .getCardByCourseAndDeck(categoryId, courseId, deckId, card.getId())).withSelfRel();
+                                                                @PathVariable Long deckId,
+                                                                @RequestParam List<String> images) {
+
+        cardService.addCard(card, deckId, images);
+        Link selfLink = linkTo(methodOn(CardController.class).getCardByCourseAndDeck(categoryId, courseId, deckId, card.getId())).withSelfRel();
         CardPublicDTO cardPublicDTO = DTOBuilder.buildDtoForEntity(card, CardPublicDTO.class, selfLink);
         return new ResponseEntity<>(cardPublicDTO, HttpStatus.CREATED);
     }
 
     @Auditable(action = AuditingAction.CREATE_CARD_VIA_CATEGORY_AND_DECK)
     @PostMapping(value = "/api/category/{categoryId}/decks/{deckId}/cards")
     @PreAuthorize(value = "@accessToUrlService.hasAccessToDeckFromCategory(#categoryId, #deckId)")
-    public ResponseEntity<CardPublicDTO> addCardByCategoryAndDeck(@Validated(Request.class) @RequestBody Card card,
+    public ResponseEntity<CardPublicDTO> addCardByCategoryAndDeck(@RequestParam String title,
```

Comment:
use @RequstBody instead all requestParams

Reply:
there are many problems with it

Reply:
That's why you are learning it, if you solve these problems it will be kind of improvement of your skills